### PR TITLE
In 146334465 validation acceptance tests

### PR DESCRIPTION
**Pull request by jgedeon-osi**

### CODE REVIEW

File: spec/shame_helpers.rb [hunk 1]
```diff
@@ -8,15 +15,28 @@ def fill_in_datepicker(locator, with:)
...
+  element = page.find_field(locator)
+  if (element.tag_name == 'select')
+    execute_script("$('##{element['id']}').trigger('focus')")
+  else
+    element.click
+  end
+end
+
+def blur
+  first('.card-body').native.click
```
Comment (AmanShurafa) [inline spec/shame_helpers.rb]: Use `first(*)` so blur can be used in limited contexts where `.card-body` isn't defined.

File: spec/screening/validations_spec.rb [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      let(:required_field_error_messages) do
+        {
+          'Assigned Social Worker' => 'Please enter an assigned worker.',
+          'Screening Start Date/Time' => 'Please enter a screening start date.',
+          'Communication Method' => 'Please select a communication method.'
+        }
+      end
+      it 'validates required fields' do
+        within '#screening-information-card' do
+          required_field_error_messages.each do |label, expected_error_message|
```
Comment (AmanShurafa) [inline spec/screening/validations_spec.rb]: 👍

File: spec/shame_helpers.rb (hunk 1 above)
Comment (jgedeon-osi) [inline spec/shame_helpers.rb]: Done

File: spec/screening/validations_spec.rb [hunk 3]
```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+require 'active_support/core_ext/numeric/time'
+
+describe 'Screening Validation QA Test', type: :feature do
+  # Selecting Create Person on homepage
```
Comment (thfong) [inline spec/screening/validations_spec.rb]: This comment does not seem accurate.

File: spec/screening/validations_spec.rb [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+          blur
+          expect(page).to_not have_content(expected_error_message)
+
+          click_button 'Save'
+
+          expect(page).to_not have_content(expected_error_message)
+        end
+      end
+
+      it 'validates start date cannot be after end date' do
```
Comment (thfong) [inline spec/screening/validations_spec.rb]: May be I missed it, but where do we address these two ACs:2. **Scenario: Missing required fields on Save**
GIVEN that Jeff, a screener, has started a new screening
WHEN Jeff has not entered data in a required field (assigned worker and/or a start date and time)
THEN Jeff can save the Screening Information card 
      BUT Jeff sees error messages 
 
3. **Scenario: Fixing missing fields**
GIVEN that Jeff did not enter data in a required field (assigned worker and/or start date and time) 
     AND is seeing error messages
WHEN Jeff enters data in the required fields
AND focuses out of the field
THEN Jeff no longer sees error messages.
*This scenario applies to both error messages displayed when the user focuses out, and when they Save.*

File: spec/screening/validations_spec.rb (hunk 3 above)
Comment (jgedeon-osi) [inline spec/screening/validations_spec.rb]: Done removed.

File: spec/screening/validations_spec.rb (hunk 4 above)
Comment (jgedeon-osi) [inline spec/screening/validations_spec.rb]: Thanks for catching this. The first scenario was covered, but the second one was missed. Corrected now.

**Pull request merged**